ecosystem/typescript/sdk/src/utils/memoize-decorator.ts

```diff
@@ -0,0 +1,144 @@
+/**
+ * Credits to https://github.com/darrylhodgins/typescript-memoize
+ */
+
```

Comment:
Could you explain how this works in greater detail? Like how this works in general, what the tags do, etc.

Reply:
Overall, I am a little reluctant to refactor this file besides a few namings (e.g. duration vs expiring). The code was copied from a library, which was battle-tested. Copy it here to reduce risks. Plus, we might add an LRU cache feature to it. But for now, I will just do minor renaming work and some comments.